scripts/pylib/pytest-twister-harness/src/twister_harness/fixtures.py

```diff
@@ -66,3 +68,23 @@ def is_mcumgr_available() -> None:
 @pytest.fixture()
 def mcumgr(is_mcumgr_available: None, dut: DeviceAdapter) -> Generator[MCUmgr, None, None]:
     yield MCUmgr.create_for_serial(dut.device_config.serial)
+
+
+@pytest.fixture(scope='session')
+def bsim_env() -> dict:
+    bsim_out_path = os.getenv('BSIM_OUT_PATH')
+    if bsim_out_path is None:
+        pytest.fail('Environmental variable BSIM_OUT_PATH has to be set.')
```

Comment:
I added this additional checking because there is possibility to run those pytest tests without Twister. And in this case, it would be valuable to check if those environment variables are set.